torchgeo/trainers/change.py

```diff
@@ -210,13 +266,34 @@ def _shared_step(self, batch: Any, batch_idx: int, stage: str) -> Tensor:
             x = rearrange(x, 'b t c h w -> b (t c) h w')
         y_hat = self(x)
 
-        loss: Tensor = self.criterion(y_hat, y.to(torch.float))
+        # For binary tasks, squeeze channel dimension to match expected shape
+        if self.hparams['task'] == 'binary':
+            y_hat = y_hat.squeeze(1)
+            y = y.squeeze(1)
+        elif self.hparams['task'] == 'multiclass':
+            y = y.squeeze(1)
+            y = y.long()
```

Comment:
I'm confused why this is needed for CD but not for segmentation. We weren't squeezing binary before this PR, why is this needed all of a sudden? I know I added an extra dimension for VideoSequential compatibility, but does that extra dimension cause issues? Is y ever not long?

Reply:
Let me clone your branch and investigate, do you mind if I push fixes to your branch?